server/src/main/java/org/tctalent/server/service/db/aws/S3ResourceHelper.java

```diff
@@ -34,292 +51,285 @@
-    }
+      // blocking wait, though we could let uploads just run in background
+      upload.waitForCompletion();
 
-    public void copyObject(String sourceBucket,
-                           String sourceKey,
-                           String destinationBucket,
-                           String destinationKey) {
+      log.debug("uploaded MultipartFile to {}", key);
 
-        sourceBucket = sourceBucket != null ? sourceBucket : s3Bucket;
-        destinationBucket = destinationBucket != null ? destinationBucket : s3Bucket;
-        CopyObjectRequest copyObjectRequest = new CopyObjectRequest(sourceBucket, sourceKey, destinationBucket,
-                                                                    destinationKey);
-        transferManager.copy(copyObjectRequest);
+    } catch (Exception e) {
+      log.error("Error uploading file to: " + key, e);
+      throw new FileUploadException("Error uploading file to: " + key, e);
     }
-
-    public void copyBucketContents(String sourceBucket,
-                                   String sourcePrefix,
-                                   String destinationBucket,
-                                   String destinationPrefix)
-            throws AmazonS3Exception {
-        ListObjectsRequest req = new ListObjectsRequest();
-        req.setBucketName(sourceBucket);
-        req.withPrefix(sourcePrefix);
-        List<S3ObjectSummary> contents = amazonS3.listObjects(req).getObjectSummaries();
-        if (contents.size() <= 0) {
-            throw new AmazonS3Exception("No files found under " + sourceBucket + " and prefix " + sourcePrefix);
-        }
-        for (S3ObjectSummary summary : contents) {
-            String newKey = destinationPrefix + summary.getKey().replace(sourcePrefix, "");
-            CopyObjectRequest copyObjectRequest = new CopyObjectRequest(sourceBucket, summary.getKey(),
-                                                                        destinationBucket, newKey);
-            transferManager.copy(copyObjectRequest);
-        }
+  }
+
+  public void uploadFile(String content,
+      String key,
+      String contentType)
+      throws FileUploadException {
+    try {
+      uploadFile(s3Bucket, content, key, contentType);
+
+    } catch (Exception e) {
+      log.error("Error uploading file to: " + key, e);
+      throw new FileUploadException("Error uploading file to: " + key, e);
     }
-
-    public File getTmpFile() throws IOException {
-        File tmpFile = File.createTempFile("lola-", ".tmp");
-        tmpFile.deleteOnExit();
-        return tmpFile;
+  }
+
+  public void uploadFile(String bucket,
+      String content,
+      String key,
+      String contentType)
+      throws FileUploadException {
+    try {
+      byte[] bytes = content.getBytes(StandardCharsets.UTF_8);
+      InputStream is = new ByteArrayInputStream(bytes);
+      uploadFile(bucket, is, bytes.length, key, contentType);
+    } catch (FileUploadException e) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: An exception is caught and re-thrown within the catch clause without any other operations being performed on the exception, making the catch clause unnecessary.

**Fix**: Perform additional operations on the exception that has been caught. You can wrap a standard exception into a custom exception to add additional information or special handling. When you do that, make sure to set the original exception as the cause. Otherwise, you will lose the stack trace and message of the original exception, which will make it difficult to analyze the event that caused the exception.  
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)